### PR TITLE
[python] bump dependencies and version

### PR DESCRIPTION
**Pull request by tadelesh**
wait for fix: https://github.com/Azure/cadl-ranch/pull/748 and new version of cadl-ranch

### CODE REVIEW

File: packages/http-client-python/package.json [hunk 1]
```diff
@@ -51,41 +51,41 @@
...
     "typescript-eslint": "^8.5.0",
-    "@azure-tools/typespec-azure-core": "~0.46.0",
-    "@azure-tools/typespec-client-generator-core": "0.46.2",
-    "@typespec/compiler": "~0.60.0",
-    "@typespec/http": "~0.60.0",
-    "@typespec/rest": "~0.60.0",
-    "@typespec/versioning": "~0.60.0",
-    "@azure-tools/typespec-azure-rulesets": "0.46.0",
+    "@azure-tools/typespec-azure-core": "~0.47.0",
+    "@azure-tools/typespec-client-generator-core": "0.47.1",
```
Comment (msyyc) [inline packages/http-client-python/package.json:83]:
```suggestion
    "@azure-tools/typespec-client-generator-core": "~0.47.1",
```
Shall we add `~` like others?

File: packages/http-client-python/package.json [hunk 2]
```diff
@@ -51,41 +51,41 @@
...
+    "@typespec/compiler": "~0.61.0",
+    "@typespec/http": "~0.61.0",
+    "@typespec/rest": "~0.61.0",
+    "@typespec/versioning": "~0.61.0",
+    "@azure-tools/typespec-azure-rulesets": "0.47.0",
```
Comment (msyyc) [inline packages/http-client-python/package.json:88]:
```suggestion
    "@azure-tools/typespec-azure-rulesets": "~0.47.0",
```
Shall we add ~ like others?

**Comment by tadelesh** [comment]
@timotheeguerin this [line](https://github.com/microsoft/typespec/blob/main/packages/website-astro/algolia.ts#L5) cause ci security scan issue which block this pr, could you help?

**Comment by timotheeguerin** [comment]
Suppressing in https://github.com/microsoft/typespec/pull/4696

**Pull request merged**